scipy/stats/tests/test_distributions.py

```diff
@@ -1669,6 +1669,29 @@ def test_rvs(self, c):
         ci = btest.proportion_ci(confidence_level=0.999)
         assert ci.low < 0.5 < ci.high
 
+    @pytest.mark.parametrize("c, ref",
+                            [(3, 0.9247941752645439),
+                             (0.001, 7.908754457315668),
+                             (5.3, 0.6164581314552009)])
+    def test_entropy(self, c, ref):
+        # Reference values were calculated with the generic method.
+        assert_allclose(stats.loggamma._entropy(c), ref, rtol=1e-14)
+
+    @pytest.mark.parametrize("c, ref",
+                            [(50, -0.5337397250497538),
+                             (2100, -2.405828413571719),
+                             (10000000000.0, -10.093986931748889),
+                             (1e+100, -113.71031611649761)])
+    def test_extreme_entropy(self, c, ref):
+        # Reference values were calculated with mpmath:
+        # from mpmath import mp
+        # mp.dps = 500
+        #
+        # def entropy(c):
+        # h = 0.5 + 0.5 * mp.log(2 * mp.pi) - 0.5 * mp.log(c) + 1 / (6 * c)
+        # return float(h)
+        assert_allclose(stats.loggamma._entropy(c), ref, rtol=1e-12)
```

Comment:
As the generic tests agree with the analytical formula you derived (all those new results you derived could be noted somewhere else than scipy also ), I think we can provide test reference values using an mpmath implementation of that formula. This is not really a test of the asymptotic formula at the moment.

Please also move all tests in just one function. If reference values are calculated in the same way for all tested values, there is no point in different testing functions in my opinion.

Reply:
>all those new results you derived could be noted somewhere else than scipy btw like on arXiv

ArXiv probably isn't the best place to put them, since it's not really something you would ever publish in a journal and I don't think you can add to the article once you submit it. That being said, I do plan to compile them at some point for easier reading, kind of like that entropy book you posted a few months ago.

>If reference values are calculated in the same way for all tested values, there is no point in different testing functions in my opinion.

They weren't calculated the same way. For the first test, I just called the entropy method before changing anything, saved it, and then copied it into the test function. Also, the tolerances are different for the 2 tests, which is probably enough justification to keep them separate.